### DISCUSSION TITLE
Expose `EditorToaster` API for plugins

### DISCUSSION THREAD

**Discussion by dalexeev** [godotengine/godot-proposals#6452] [Editor]
Toast notifications can be a good alternative to `OS.alert()` (forces the user to click OK) and `push_error()`/`push_warning()` (may not be noticed by the user).

**Comment by ashifolfi** [comment]
I heavily agree with this! Working with someone who is a non coder and gets very confused with programming related things I found that there's no real easy way for me to display notices from my graphical editors to them without dialogs or making them go into a debug log.

having the option to make push notifications for these things would be extremely useful to me in my mission to make my friend feel more directly involved with the development of the project without overwhelming them.

**Comment by Calinou** [comment]
We could make the `push_error()` and `push_warning()` exposed to GDScript use `ERR_PRINT_ED` and `WARN_PRINT_ED` instead of `ERR_PRINT` and `WARN_PRINT` respectively. This way, all usage of `push_error()` and `push_warning()` within editor plugins or `@tool` scripts will display toasts in the editor.

This wouldn't make it possible to spawn informational toasts though (`SEVERITY_INFO`) – not until we implement https://github.com/godotengine/godot-proposals/issues/1378.

https://github.com/godotengine/godot/blob/e7d2e49688bdaec65e326e7b2bfcc0642b4f2a8d/core/variant/variant_utility.cpp#L905

https://github.com/godotengine/godot/blob/e7d2e49688bdaec65e326e7b2bfcc0642b4f2a8d/core/variant/variant_utility.cpp#L925

**Edit:** Pull request opened: https://github.com/godotengine/godot/pull/77890

**Comment by Calinou** [comment]
Proposal issue opened: https://github.com/godotengine/godot-proposals/issues/11051
